### PR TITLE
Fix Unaware Confusion interaction

### PR DESCRIPTION
**Pull request by jmclemo6**
This is my first time working with actual battle code and writing a test, so if something is wrong let me know.

### CODE REVIEW

File: data/statuses.js [hunk 1]
```diff
@@ -171,6 +171,7 @@ exports.BattleStatuses = {
 			if (this.random(3) > 0) {
 				return;
 			}
+			this.activeTarget = this;
```
Comment (urkerab) [inline data/statuses.js]: Did you mean `this.activeTarget = pokemon;`?

Comment (jmclemo6) [inline data/statuses.js]: Yes, I did. Thank you.

File: test/simulator/abilities/unaware.js [hunk 2]
```diff
@@ -78,4 +78,18 @@ describe('Unaware', function () {
 		battle.commitDecisions();
 		assert.notStrictEqual(pokemon.maxhp - pokemon.hp, damage);
 	});
+
+	it('should not affect opponent\'s self hit damage from confusion', function () {
```
Comment (QuiteQuiet) [inline test/simulator/abilities/unaware.js]: This test doesn't actually test this mechanic at all. It will pass without any changes to Unaware at all.

Comment (jmclemo6) [inline test/simulator/abilities/unaware.js]: Alright, I removed the test for now.

**Pull request merged**